packages/firestore/src/local/shared_client_state.ts

```diff
@@ -773,6 +776,10 @@ export class WebStorageSharedClientState implements SharedClientState {
           return;
         }
 
+        if (event.key === null) {
```

Comment:
Shouldn't this be `==`? (We test `event.newValue != null` just below.)

Reply:
I'm surprised that the linter didn't complain about `event.newValue != null`. We should not rely on type coercion, and I would guess that implicit coercion is almost always unintended. So rather than using `==` here, I changed the line below to `!==`. The type for LocalStorage are defined as "string | null", so they should never be `undefined`.

Reply:
I think the issue is that `== null` is the one case where double-equal is actually useful because it handles `undefined` in a sane way. I don't actually know which we prefer, but ISTR this point being discussed.